query-engine/connector-test-kit/src/test/scala/writes/unchecked_writes/UncheckedNestedCreateSpec.scala

```diff
@@ -0,0 +1,415 @@
+        |  createOneModelB(data: {
+        |    uniq_1: 2
+        |    uniq_2: 2
+        |    a: {
+        |      create: {
+        |        id: 2
+        |      }
+        |    }
+        |  }) {
+        |    a {
+        |      b {
+        |       uniq_1
+        |       uniq_2
+        |      }
+        |    }
+        |  }
+        |}
+      """.stripMargin,
+      project,
+      legacy = false
+    )
+
+    res.toString() should be("""{"data":{"createOneModelB":{"a":[{"b":{"uniq_1":2,"uniq_2":2}}]}}}""")
+
+    res = server.query(
+      """
+        |mutation {
+        |  createOneModelB(data: {
+        |    uniq_1: 3
+        |    uniq_2: 3
+        |    a: {
+        |      create: {
+        |        id: 3
+        |        c_id_1: null
+        |        c_id_2: 123
+        |      }
+        |    }
+        |  }) {
+        |    a {
+        |      b {
+        |       uniq_1
+        |       uniq_2
+        |      }
+        |      c {
+        |        uniq_1
+        |        uniq_2
+        |      }
+        |    }
+        |  }
+        |}
+      """.stripMargin,
+      project,
+      legacy = false
+    )
+
+    res.toString() should be("""{"data":{"createOneModelB":{"a":[{"b":{"uniq_1":3,"uniq_2":3},"c":null}]}}}""")
+  }
+
+  "Unchecked nested creates" should "fail if required relation scalars are not provided" in {
+    val project = ProjectDsl.fromString {
+      """|model ModelA {
+         |  id     Int  @id
+         |  b_id_1 Int
+         |  b_id_2 Int
+         |  c_id_1 Int
+         |  c_id_2 Int
+         |
+         |  b ModelB @relation(fields: [b_id_1, b_id_2], references: [uniq_1, uniq_2])
+         |  c ModelC @relation(fields: [c_id_1, c_id_2], references: [uniq_1, uniq_2])
+         |}
+         |
+         |model ModelB {
+         |  uniq_1    Int
+         |  uniq_2    Int
+         |  
+         |  a ModelA[]
+         |
+         |  @@unique([uniq_1, uniq_2])
+         |}
+         |
+         |model ModelC {
+         |  uniq_1    Int
+         |  uniq_2    Int
+         |  
+         |  a ModelA[]
+         |
+         |  @@unique([uniq_1, uniq_2])
+         |}
+      """
+    }
+
+    database.setup(project)
+
+    server.queryThatMustFail(
+      """
+        |mutation {
+        |  createOneModelB(data: {
+        |    uniq_1: 1
+        |    uniq_2: 1
+        |    a: {
+        |      create: {
+        |        id: 1
+        |        c_id_1: 123,
+        |      }
+        |    }
+        |  }) {
+        |    uniq_1
+        |    uniq_2
+        |  }
+        |}
+      """.stripMargin,
+      project,
+      errorCode = 2009,
+      errorContains =
+        """`Mutation.createOneModelB.data.ModelBUncheckedCreateInput.a.ModelAUncheckedCreateManyWithoutBInput.create.ModelAUncheckedCreateWithoutBInput.c_id_2`: A value is required but not set.""",
+      legacy = false
+    )
+  }
+
+  "Unchecked nested creates" should "not allow writing inlined relations regularly" in {
```

Comment:
This description is not very intuitive to me. Since you introduced the term "Unchecked" for this API can we maybe call the regular API something like "Checked Relation API"?

Reply:
Even if you have an unchecked type, you can still write other relations that are NOT inlined regularly.